packages/cli/src/services/abstract-check-runner.ts

```diff
@@ -32,55 +29,34 @@ export type PublicRunLocation = {
 }
 export type RunLocation = PublicRunLocation | PrivateRunLocation
 
-type CheckRunId = string
+export type CheckRunId = string
 
-export default class CheckRunner extends EventEmitter {
-  project: Project
-  checks: Map<CheckRunId, any>
-  groups: Record<string, CheckGroup>
+export default abstract class AbstractCheckRunner extends EventEmitter {
+  checks?: Map<CheckRunId, any>
```

Comment:
That sounds good 👍

Reply:
I tried it out, and it doesn't quite work.

`configureResultListener` would need the `checks` in order to configure the `socketClient.on('message')`. Since we need to call `configureResultListener` before `scheduleChecks`, we can't pass `checks` around as a function parameter.

Reply:
It's not possible to pass `checks` as a function parameter here https://github.com/checkly/checkly-cli/pull/658/files#diff-2553333456f01ab3b0c2d63712b790a4e771cc4a8ddf433c5b8a93b8545e86cdR103

Reply:
Gotcha